csm/core/controllers/rgw/s3/users.py

```diff
@@ -0,0 +1,81 @@
+# CORTX-CSM: CORTX Management web and CLI interface.
+# Copyright (c) 2022 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import json
+from marshmallow import Schema, fields, ValidationError, validate, validates_schema
+from cortx.utils.log import Log
+from csm.common.errors import InvalidRequest
+from csm.common.permission_names import Resource, Action
+from csm.core.blogic import const
+from csm.core.controllers.view import CsmView, CsmAuth, CsmResponse
+from csm.core.controllers.validators import ValidationErrorFormatter
+from csm.core.controllers.rgw.s3.base import S3BaseView
+
+class UserCreateSchema(Schema):
+    """S3 IAM User create schema validation class."""
+
+    uid = fields.Str(data_key=const.UID, required=True)
+    display_name = fields.Str(data_key=const.DISPLAY_NAME, required=True)
+    email = fields.Email(data_key=const.EMAIL, missing=None)
+    key_type = fields.Str(data_key=const.KEY_TYPE, missing=None,
+                    validate=validate.OneOf(['s3']))
+    access_key = fields.Str(data_key=const.ACCESS_KEY, missing=None)
+    secret_key = fields.Str(data_key=const.SECRET_KEY, missing=None)
+    user_caps = fields.Str(data_key=const.USER_CAPS, missing=None)
+    generate_key = fields.Bool(data_key=const.GENERATE_KEY, missing=None)
+    max_buckets = fields.Int(data_key=const.MAX_BUCKETS, missing=None)
+    suspended = fields.Bool(data_key=const.SUSPENDED, missing=None)
+    tenant = fields.Str(data_key=const.TENANT, missing=None)
+
+    @validates_schema
+    def invalidate_empty_values(self, data, **kwargs):
+        """This method invalidates the empty strings."""
+        for key, value in data.items():
+            if value is not None and not str(value).strip():
```

Comment:
Minor: could `validate=validate.Length(min=1)` do this job for you?

Reply:
Actually No, as validate=validate.Length(min=1) allows {"uid":"   "}